### DISCUSSION TITLE
Proposal: Implementing MySQL Support in Hyperswitch

### DISCUSSION THREAD

**Discussion by Pavanmm1762** [juspay/hyperswitch#4183] [General]
Objective:
The objective of this proposal is to extend the database support of Hyperswitch from PostgreSQL (PSQL) to include MySQL, one of the most widely used relational database management systems, in order to broaden compatibility and meet the needs of a larger user base.

Background:
Currently, Hyperswitch utilizes the Diesel ORM library for database communication, with support for PSQL out of the box. However, given MySQL's significant market share in the relational database market, it is imperative to ensure compatibility with MySQL for wider adoption of Hyperswitch.

Approach:

Configuration Update:

Modify the Diesel configuration to add support for MySQL alongside PSQL. This involves updating connection strings and any other relevant configurations to accommodate MySQL.
Testing and Validation:

Conduct thorough testing to ensure seamless integration with MySQL as the backend database.
Test various functionalities of Hyperswitch, including payment processing and data retrieval, to validate compatibility and functionality with MySQL.
Addressing Compatibility Issues:

Identify and address any compatibility issues or discrepancies encountered during testing.
Modify database queries, schema definitions, or any other relevant components to ensure consistent behavior across both PSQL and MySQL.
Performance Optimization:

Evaluate the performance of Hyperswitch with MySQL and identify any potential performance bottlenecks.
Implement optimizations and best practices to improve the performance and efficiency of Hyperswitch when using MySQL as the backend database.
Timeline:

Week 1: Configure Diesel for MySQL and conduct initial testing.
Week 2: Address compatibility issues and optimize performance.
Week 3: Finalize testing and validation, prepare for deployment.
Week 4: Deploy MySQL support to production environment and monitor for any issues or regressions.
Deliverables:

Updated Diesel configuration with MySQL support.
Thorough testing documentation, including test cases and results.
Performance optimization strategies and implementation details.
Deployment plan and documentation for adding MySQL support to Hyperswitch in production.
Conclusion:
By implementing MySQL support in Hyperswitch, we aim to enhance the flexibility and compatibility of the platform, enabling a broader range of users to leverage its capabilities. This proposal outlines a structured approach to achieve this objective in a systematic and efficient manner.

**Comment by neerajkumar-n** [comment]
@Pavanmm1762 why don't you share the proposal within the discussion > comments. 

Discussion link: https://github.com/juspay/hyperswitch/discussions/4179
